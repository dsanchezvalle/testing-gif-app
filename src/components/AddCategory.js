import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
        //console.log('Handle change activated')
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        //console.log('Handle Submit called', inputValue)
        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
