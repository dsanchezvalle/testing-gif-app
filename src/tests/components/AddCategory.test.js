import React from 'react';
import { AddCategory } from '../../components/AddCategory'
import { shallow } from 'enzyme'

describe('AddCategory tests', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}  />);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}  />);
    });

    test('AddCategory should render properly', ()=>{
        expect(wrapper).toMatchSnapshot();
    });

    test('input should change', ()=>{
        const input = wrapper.find('input');
        const value = "Hello world!";
        input.simulate('change', {target: {value}});
        const inputAfter = wrapper.find('input');
        expect(inputAfter.prop('value')).toBe(value);
        //You can test with an extra paragraph that the value changed.
    });

    test('it should not post info on submit', () => {
        //const prevDef = jest.fn();

        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('it should call setCategories and reset input', ()=>{
        const value = 'Test value';
        wrapper.find('input').simulate('change', {target:{value}});
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        //expect(setCategories).toHaveBeenCalledTimes(2);
        expect(wrapper.find('input').prop('value')).toBe('');

    })

});