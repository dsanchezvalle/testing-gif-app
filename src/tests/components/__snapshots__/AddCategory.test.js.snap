// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory tests AddCategory should render properly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
